get_rt.Rmd

```diff
@@ -0,0 +1,187 @@
+---
+title: "Define the retention times"
+author: "Mar Garcia-Aloy"
+output: 
+  BiocStyle::html_document:
+    toc: true
+    number_sections: false
+    toc_float: true
+---
+
+```{r settings, echo = FALSE, results = "hide", message = FALSE}
+study <- "standards_dilution" # specify "internal_standards" OR 
+                              #         "standards_dilution"
+mixnum <- 17 # specify which MIX
+polarity <- "POS" # specify "POS" or "NEG"
+da <- 0.01
+
+#' Define the path where we can find the mzML files:
+MZML_PATH <- "C:/Users/mgarciaaloy/Documents/mzML/"
+#if (!file.exists(MZML_PATH))
+#    stop("Can not find the directory with the mzML files: ", MZML_PATH)
+```
+
+# Preliminaries 
+```{r, message=FALSE}
+library(xcms)
+library(magrittr)
+library(CompoundDb)
+library(Rdisop)
+```
+
+# Data import
+In the file `XXXX_files.txt` there is the information regarding 
+the injection sequence.
+```{r}
+injections <- read.table(paste0("data/", study, "_files.txt"), #import the file
+                         sep = "\t", header = TRUE, as.is = TRUE)
+myfiles <- injections$mzML # get file names
+myfiles <- myfiles[grep(polarity, myfiles)]# select files names of our polarity
+if(study == "standards_dilution"){
+  myfiles <- myfiles[grep(paste0("MIX ", mixnum, "K"), myfiles)]
+  }
+myfiles
+```
+
+Import the information regarding the standards that are in the samples.  
+```{r}
+std_info <- read.table(paste0("data/", study, ".txt"),
+                      sep = "\t", header = TRUE, as.is = TRUE)
+std_info$name <- c(substring(std_info$name, 1, 33))
+if(study == "standards_dilution"){
+  std_info <- subset(std_info, mix == mixnum)
+}
+std_info$mzneut = NA
+for(i in seq(nrow(std_info))){
+  if(grepl("C", std_info$formula[i])){std_info$mzneut[i] = 
+    getMolecule(as.character(std_info$formula[i]))$exactmass}else{
+      std_info$mzneut[i] = paste(std_info$formula[i])}
```

Comment:
Also here, please use `as.numeric` instead of `paste` and remove the code below exporting/importing to/from csv file.